js/widgets/widget.backcompat.js

```diff
@@ -51,50 +63,62 @@ if ( $.mobileBackcompat !== false ) {
 			}
 		},
 
-		_optionsToClasses: function( option ) {
+		_optionsToClasses: function( option, value ) {
 			var classArray,
-				newValue = "",
 				prop = this.classProp,
 				classes = this.options.classes,
 				className = this._boolOptions[ option ];
 
 			if ( prop ) {
-				classArray = classes[ prop ].split ( " " );
+				classArray = classes[ prop ] ? classes[ prop ].split ( " " ) : [];
 
-				if ( this.options[ option ] ) {
-					newValue = classes[ prop ] + " " + className;
+				if ( value ) {
+					classArray.push( className );
 				} else {
-					newValue = classArray
-						.splice( classArray.indexOf( this._boolOptions[ option ] ) - 1, 1 )
-						.join( " " );
+					classArray = removeOneElementFromArray( classArray,
+							classArray.indexOf( this._boolOptions[ option ] ) );
 				}
-				this.option( "classes." + prop, newValue );
+				this.option( "classes." + prop, classArray.join( " " ) );
 			}
 		},
 
-		_setInitalOptions: function() {
-			var originalClasses, currentClasses,
-				that = this,
+		_setInitialOptions: function() {
+			var currentClasses,
 				options = this.options,
 				original = $[ this.namespace ][ this.widgetName ].prototype.options,
 				prop = this.classProp;
 
-			if ( prop && original.classes[ prop ] && this.options.classes[ prop ] ) {
-				originalClasses = original.classes[ prop ].split( " " );
+			if ( prop && typeof original.classes[ prop ] === "string" &&
+					typeof this.options.classes[ prop ] === "string" ) {
 				currentClasses = this.options.classes[ prop ].split( " " );
 
-				$.each( this._boolOptions, function( option, className ) {
-					if( that.options[ option ] !== undefined ) {
-						var initial = ( originalClasses.indexOf( className ) !== -1 ),
-							current = ( currentClasses.indexOf( className ) !== -1 );
+				// If the classes option value has diverged from the default, then its value takes
+				// precedence, causing us to update all the style options to reflect the contents
+				// of the classes option value
+				if ( original.classes[ prop ] !== this.options.classes[ prop ] ) {
+					$.each( this._boolOptions, function( option, className ) {
+						if( options[ option ] !== undefined ) {
+							options[ option ] = ( currentClasses.indexOf( className ) !== -1 );
+						}
+					}) ;
 
-						if ( initial !== current ) {
-							options[ option ] = current;
-						} else if ( options[ option ] !== original[ option ] ) {
-							that._optionsToClasses( option, options[ option ] );
+				// Otherwise we assume that we're dealing with legacy code and look for style
+				// option values which diverge from the defaults, and update the classes option
+				// value accordingly
```

Comment:
This is not true there is the third case where they are both default where we should do nothing.